### ISSUE TITLE
Needing a LoRA-weight merging script for real-world use

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/lit-gpt#494]
I mentioned before that we could really use a script that merges LoRA weights with the base model to get an updated checkpoint model (tagging you @carmocca @JustinGoheen since I think we discussed this before, and I wanted to raise an additional point in favor of such a script).

The script's usage could be 

```bash
python finetune/lora.py \ # as usual
--checkpoint_dir base_model
--out_dir lora_weights

python scripts/merge_lora_weights.py \ # new
--checkpoint_dir base_model
--lora_weights lora_weights
--out_dir new_checkpoint
```

## Overcoming finetuning limitations

As far as I know, it's currently not possible to finetune a model on multiple datasets due to the following limitation:

```bash
python finetune/lora.py \
--checkpoint_dir basemodel \
--data dir dolly \
--out_dir lora_weights_1

python finetune/lora.py \
--checkpoint_dir ??? \
--data dir lima \
--out_dir lora_weights_2
```

Right now, finetuning on multiple datasets would only be possible with adapters, as far as I know?

```bash
python finetune/adapter.py \
--checkpoint_dir basemodel \
--data dir dolly \
--out_dir finetuned_model_stage_1

python finetune/adapter.py \
--checkpoint_dir finetuned_model_stage_1 \
--data dir lima \
--out_dir finetuned_model_stage_2
```

## Simplifying the repository

Also, if we had such a script, we wouldn't need to maintain separate evaluation scripts like [lm_eval_harness.py](https://github.com/Lightning-AI/lit-gpt/blob/main/eval/lm_eval_harness.py) and [lm_eval_harness_lora.py](https://github.com/Lightning-AI/lit-gpt/blob/main/eval/lm_eval_harness_lora.py)

**Comment by carmocca** [comment]
@rasbt Now that #495 landed, the suggested simplification is possible: removing `generate/lora.py` and `eval/lm_eval_harness_lora.py`

**Comment by rasbt** [comment]
I think we can remove `eval/lm_eval_harness_lora.py` (I think it's broken anyway). I can do that in a separate PR and update the docs.

But I am a bit hesitant about `generate/lora.py` because not everyone wants to merge weights. I think it's a trade-off between storage and inference compute. Keeping `generate/lora.py` might make sense.

**Comment by rasbt** [comment]
This is going to be addressed in #532
